cirq/linalg/decompositions.py

```diff
@@ -264,12 +264,115 @@ def so4_to_magic_su2s(
     return a, b
 
 
-def kak_canonicalize_vector(
-        x: float, y: float, z: float
-) -> Tuple[complex,
-           Tuple[np.ndarray, np.ndarray],
-           Tuple[float, float, float],
-           Tuple[np.ndarray, np.ndarray]]:
+class KakDecomposition:
+    """A convenient description of an arbitrary two-qubit operation.
+
+    Any two qubit operation U = can be decomposed into the form
+
+        U = g · (a1 ⊗ a0) · exp(i·(x·XX + y·YY + z·ZZ)) · (b1 ⊗ b0)
+
+    This class stores g, (b0, b1), (x, y, z), and (a0, a1).
+
+    Attributes:
+        global_phase: g from the above equation.
+        single_qubit_operations_before: b0, b1 from the above equation.
+        interaction_coefficients: x, y, z from the above equation.
+        single_qubit_operations_after: a0, a1 from the above equation.
+    """
+
+    def __init__(self,
+                 *,
+                 global_phase: complex,
+                 single_qubit_operations_before: Tuple[np.ndarray, np.ndarray],
+                 interaction_coefficients: Tuple[float, float, float],
+                 single_qubit_operations_after: Tuple[np.ndarray, np.ndarray]):
+        """Initializes a decomposition for a two-qubit operation U.
```

Comment:
This looks the same as the class pydoc.  Refer to that instead?

Reply:
Acknowledged. They may appear in different places (e.g. based on what is being autocompleted in an IDE, or if the user is making-vs-consuming a KakDecomposition), so I think the repetition is worth it in this case. My opinion is likely to change as I look into the finer details of how our documentation is generated.